projects/observability/src/shared/dashboard/data/graphql/explore/explore-result.ts

```diff
@@ -64,10 +68,24 @@ export class ExploreResult {
     return this.resultsContainingSpec(spec).map(result => this.resultToGroupData(result, groupBySpecs, spec));
   }
 
+  private extractMetricAggregationForSpec(spec: ExploreSpecification): MetricAggregationData {
+    return this.resultToMetricAggregationDataData(this.resultsContainingSpec(spec)?.[0]!, spec);
```

Comment:
We're asserting here a result will always be returned - what happens if we don't get any data back? Not sure if that can happen or not (i.e. does the server default if no data in range for aggregation), but seems like we can handle it regardless to be safe.

Reply:
So in the case that this doesn't return data we can return an empty array?

Reply:
Yeah I think you'd ultimately want to render "No data" - so that probably translates to no result series.

Reply:
Alternatively a series with an empty data array could make sense too

Reply:
what if I return the value as 0 in this case? I believe that is an unreachable scenario but we could handle this scenario returning 0 as value.

Reply:
I'd aim for the empty version (either no series or series of empty data) - a zero and a null (no result) are sometimes equivalent but not always, it depends on the query. For example if I ask "how many calls occurred in the last 1h" showing either 0 or no data makes sense. If I ask "What is the average latency for calls in the last hour" then 0 is not correct - we'd want no data. 

Hope that makes sense

Reply:
done